hazelcast/src/main/java/com/hazelcast/map/impl/MapRemoteService.java

```diff
@@ -44,8 +47,15 @@
     @Override
     public DistributedObject createDistributedObject(String name) {
         MapConfig mapConfig = nodeEngine.getConfig().findMapConfig(name);
+        checkInMemoryFormat(mapConfig.getInMemoryFormat());
```

Comment:
Enterprise has its own proxies, means no call to oss `createDistributedObject` method, so there should not be any problem.